### ISSUE TITLE
Image FEC loss improvements

### ISSUE THREAD

**Issue by cwbeitel** [projectclarify/pcml#67]
Our [current facial expression understanding model](https://github.com/projectclarify/pcml/blob/master/pcml/models/percep_similarity_emb.py#L194) performs well enough to be used for feedback and provides good qualitative results. Here we consider possible improvements to losses used in training this model.

For training on both the FEC expression similarity problem and pre-training on the VGGFace2 identity problem we use the triplet loss (1) advocated by Schroff, Kalenichenko, and Philbin (2015), reformulated by Agarwala and Vemulapalli (2019) to consider either I_1 or I_2 as anchor points (2), both encouraging the distance between positive pairs to be less than negative ones. The training time loss for SGD is computed in the standard way as the sum of the element-wise losses over the batch normalized by the batch size.

![image](https://user-images.githubusercontent.com/2348401/69824916-35abca80-11c2-11ea-8bf6-762f7407119d.png)

Here d(e_i,e_j) is the L2 distance and 𝛿 is a margin of 0.1. Geometrically, for a single triplet, the latter can be imagined (Figure 1) as the maximum intersection two spherical functions (taking larger values at their center) of radius d(e_I_1,e_I_2) around e_I_1 and e_I_2 plus a 𝛿-width envelope. The loss is zero anywhere outside of this domain.

![image](https://user-images.githubusercontent.com/2348401/69824962-668bff80-11c2-11ea-9c89-ec90289c3c07.png)

Figure 1. Given two similar points in embedding space (blue) the direction of optimization encouraged by the loss function is illustrated for an initial (red; higher loss) and subsequent (yellow; lower loss) up to a loss value of zero (green) beyond a 𝛿-width envelope of the domain. The effect optimization pressure on elements within different sub-domain is contrasted for the single-anchor (1) and dual-anchor (2) loss functions illustrating the added informativeness of the latter.

A pairwise distance other than the euclidean distance is a good place to start as is a geometric intuition about possible issues and solutions with this and possible alternative losses.
